pkg/controller/petset/pet_set.go

```diff
@@ -343,6 +343,11 @@ func (psc *PetSetController) syncPetSet(ps *apps.PetSet, pets []*api.Pod) (int,
 			if err == nil {
 				numPets++
 			}
+			switch err.(type) {
+			case *unhealthyPetBlock:
```

Comment:
Error interface has a pointer receiver for Error method, but I can put it in the `else` clause (to make sure not to dereference `nil`) and check the dereferenced `err` type.